flood_adapt/dbs_controller.py

```diff
@@ -748,16 +813,35 @@ def delete_scenario(self, name: str):
         ValueError
             Raise error if scenario has already model output
         """
-        scenario_path = self.input_path / "scenarios" / name
-        scenario = Scenario.load_file(scenario_path / f"{name}.toml")
-        scenario.init_object_model()
-        if scenario.direct_impacts.hazard.has_run:
-            # TODO this should be a check were if the scenario is run you get a warning?
+
+        # Get all the benefits
+        benefits = [Benefit.load_file(path) for path in self.get_benefits()["path"]]
+
+        # Check in which benefits this scenario is used
+        used_in_benefit = [
+            benefit.attrs.name
+            for benefit in benefits
+            if name
+            in self.check_benefit_scenarios(benefit)["scenario created"].to_list()
+        ]
+
+        # If strategy is used in a benefit, raise error
+        if used_in_benefit:
+            text = "benefit" if len(used_in_benefit) == 1 else "benefits"
             raise ValueError(
-                f"'{name}' scenario cannot be deleted since the hazard model has already run."
+                f"'{name}' strategy cannot be deleted since it is already used in {text}: {', '.join(used_in_benefit)}"
             )
         else:
-            shutil.rmtree(scenario_path, ignore_errors=True)
+            scenario_path = self.input_path / "scenarios" / name
+            scenario = Scenario.load_file(scenario_path / f"{name}.toml")
+            scenario.init_object_model()
+            if scenario.direct_impacts.hazard.has_run:
+                # TODO this should be a check were if the scenario is run you get a warning?
+                raise ValueError(
+                    f"'{name}' scenario cannot be deleted since the hazard model has already run."
```

Comment:
maybe better return "the scenario has been run" instead of "the hazard model has been run"